### PR TITLE
Add notification commands

### PR DESCRIPTION
**Pull request by hrchu**
image

Sepc for ref:
https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketNotificationConfiguration.html#API_PutBucketNotificationConfiguration_RequestSyntax
https://docs.aws.amazon.com/AmazonS3/latest/userguide/NotificationHowTo.html

### CODE REVIEW

**Comment by hrchu** [comment]
@fviard anything I can help with this?

**Comment by hrchu** [comment]
@fviard anything I can help with this?

File: s3cmd [hunk 1]
```diff
@@ -2135,6 +2135,47 @@ def cmd_dellifecycle(args):
...
+    response = s3.set_notification_policy(uri, notification_policy)
+
+    debug(u"response - %s" % response['status'])
+    if response['status'] == 200:
+        output(u"%s: Notification Policy updated" % uri)
+    return EX_OK
+
+def cmd_getnotification(args):
+    cfg = Config()
+    s3 = S3(cfg)
```
Comment (fviard) [inline s3cmd:2159]: If cfg is not used inside the function, you can simplify the code by having:   
`s3 = S3(Config())`

I appreciate that you tried to respect what was  already donein the rest of the code, but it is more legacy of non clean code than a coding style :-)

File: s3cmd [hunk 2]
```diff
@@ -2135,6 +2135,47 @@ def cmd_dellifecycle(args):
...
+    uri = S3Uri(args[0])
+
+    response = s3.get_notification_policy(uri)
+
+    output(u"%s" % getPrettyFromXml(response['data']))
```
Comment (fviard) [inline s3cmd:2164]: I might be wrong, but I don't think that you need the `u"%s" %` if you just want to output the `getPrettyFromXml(...)`.

**Comment by fviard** [comment]
@hrchu Thank you for your PR, it is a very nice addition to s3cmd.  
I'm very sorry for the long delay I took to review your PR.

I did two very minor suggestions of changes, would it be possible for you to have a look at that?  
I promise to merge right away after that :-)

**Comment by hrchu** [comment]
@fviard thank you for your review and suggestions. The PR has been updated plz have a look.

I am grateful for what you've done on this project. I just wanted to make sure I've done my responsibility as a contributor so you can review this PR with less effort.

**Comment by fviard** [comment]
Merged, thank you!

**Pull request merged**